wallet/src/wallet_events.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2023 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use wallet_types::{AccountWalletTxId, WalletTx};
+
+pub trait WalletEvents {
```

Comment:
Please document the purpose of this... What problem it solves.

I'm trying to understand why everything here is `&mut self`, too.

Reply:
Comments added. `&mut self` is not a problem here, so I don't see why we should change it.